lib/Conversion/TorchToTosa/TosaLegalizeCommon.cpp

```diff
@@ -27,6 +28,370 @@
 namespace mlir {
 namespace tosa {
 
+using namespace mlir::torch::Torch;
+
+llvm::Optional<Value>
+createOneDimTfIndices(PatternRewriter &rewriter, Operation *op,
+                      SmallVector<int64_t> indicesOneDimShape, int32_t dim,
+                      ArrayRef<int64_t> indexShape) {
+  unsigned indexRank = indexShape.size();
+  SmallVector<int32_t> indicesVec;         // input vec to create tosaConstant
+  SmallVector<int32_t> indicesMetaElement; // torch.meshgrid inputs
+  int indicesMetaElementRepeatTimes{1};    // For torch.stack(torch.meshgrid)
+
+  // Create torch.meshgrid inputs
+  // Example: indexShape=[1,4,2]
+  // dim0: indicesMetaElement = torch.arange(0, 1) = [0]
+  // dim1: indicesMetaElement = torch.arange(0, 4) = [0,1,2,3]
+  // dim2: indicesMetaElement = torch.arange(0, 2) = [0,1]
+  for (int i = 0; i < indexShape[dim]; i++) {
+    indicesMetaElement.push_back(i);
+  }
+
+  // Compute total number of meta element repeat times:
+  // = product(indexShape[0:dim]) x product(indexShape[dim+1:-1]), skip dim
+  // dim0: indicesMetaElementRepeatTimes = 1      x 4*2 = 8
+  // dim1: indicesMetaElementRepeatTimes = 1 *1   x   2 = 2
+  // dim2: indicesMetaElementRepeatTimes = 1 *1*4       = 4
+  for (int i = 0; i < indexRank; i++) {
+    if (i == dim) {
+      continue;
+    } else {
+      indicesMetaElementRepeatTimes *= indexShape[i];
+    }
+  }
+
+  if (dim != indexShape.size() -
+                 1) { // Create one dim indices for index except for last dim
```

Comment:
the indentation is off here. Use `git clang-format`

Reply:
Done